crates/bevy_core_pipeline/src/tonemapping/tonemapping_shared.wgsl

```diff
@@ -1,5 +1,238 @@
+}
+
+// --------------------------------
+// ------------- AgX --------------
+// --------------------------------
+// https://github.com/MrLixm/AgXc
+// https://github.com/sobotka/AgX
+
+// pow() but safe for NaNs/negatives
+fn powsafe(color: vec3<f32>, power: f32) -> vec3<f32> {
+    return pow(abs(color), vec3(power)) * sign(color);
+}
+
+/*
+    Increase color saturation of the given color data.
+    :param color: expected sRGB primaries input
+    :param saturationAmount: expected 0-1 range with 1=neutral, 0=no saturation.
+    -- ref[2] [4]
+*/
+fn saturation(color: vec3<f32>, saturationAmount: f32) -> vec3<f32> {
+    let luma = tonemapping_luminance(color);
+    return mix(vec3(luma), color, vec3(saturationAmount));
+}
+
+/*
+    Output log domain encoded data.
+    Similar to OCIO lg2 AllocationTransform.
+    ref[0]
+*/
+fn convertOpenDomainToNormalizedLog2(color: vec3<f32>, minimum_ev: f32, maximum_ev: f32) -> vec3<f32> {
+    let in_midgrey = 0.18;
+
+    // remove negative before log transform
+    var color = max(vec3(0.0), color);
+    // avoid infinite issue with log -- ref[1]
+    color = select(color, 0.00001525878 + color, color  < 0.00003051757);
+    color = clamp(
+        log2(color / in_midgrey),
+        vec3(minimum_ev),
+        vec3(maximum_ev)
+    );
+    let total_exposure = maximum_ev - minimum_ev;
+
+    return (color - minimum_ev) / total_exposure;
+}
+
+// Inverse of above
+fn convertNormalizedLog2ToOpenDomain(color: vec3<f32>, minimum_ev: f32, maximum_ev: f32) -> vec3<f32> {
+    var color = color;
+    let in_midgrey = 0.18;
+    let total_exposure = maximum_ev - minimum_ev;
+
+    color = (color * total_exposure) + minimum_ev;
+    color = pow(vec3(2.0), color);
+    color = color * in_midgrey;
+
+    return color;
+}
+
+
+/*=================
+    Main processes
+=================*/
+
+// Prepare the data for display encoding. Converted to log domain.
+fn applyAgXLog(Image: vec3<f32>) -> vec3<f32> {
+    var Image = max(vec3(0.0), Image); // clamp negatives
+    let r = dot(Image, vec3(0.84247906, 0.0784336, 0.07922375));
+    let g = dot(Image, vec3(0.04232824, 0.87846864, 0.07916613));
+    let b = dot(Image, vec3(0.04237565, 0.0784336, 0.87914297));
+    Image = vec3(r, g, b);
+
+    Image = convertOpenDomainToNormalizedLog2(Image, -10.0, 6.5);
+    
+    Image = clamp(Image, vec3(0.0), vec3(1.0));
+    return Image;
+}
+
+fn applyLUT3D(Image: vec3<f32>, block_size: f32) -> vec3<f32> {
+    return sample_current_lut(Image * ((block_size - 1.0) / block_size) + 0.5 / block_size).rgb;
+}
+
+// -------------------------
+// -------------------------
+// -------------------------
+
+fn sample_blender_filmic_lut(stimulus: vec3<f32>) -> vec3<f32> {
+    let block_size = 64.0;
+    var c = stimulus;
+    let normalized = saturate(convertOpenDomainToNormalizedLog2(stimulus, -11.0, 12.0));
+    return applyLUT3D(normalized, block_size);
+}
+
+
+fn rgb_to_srgb_simple(color: vec3<f32>) -> vec3<f32> {
```

Comment:
Added a comment about using this in the dithering code, upon closer inspection it looks like this function isn't used by anything. Is this intentional?

Reply:
Ah it was probably used in the aces tonemapping in the bloom PR that got removed. But I think it was there even before that.